usaspending_api/data_load/fpds_loader.py

```diff
@@ -0,0 +1,299 @@
+                    "where detached_award_proc_unique = '{}'".format(
+                        load_object["transaction_fpds"]["detached_award_proc_unique"]
+                    )
+                )
+                cursor.execute(find_matching_transaction_sql)
+                results = cursor.fetchall()
+
+                if results:
+                    # If there is a transaction (transaction_normalized and transaction_fpds should be one-to-one)
+                    # we update all values
+                    columns, values, pairs = setup_load_lists(load_object, "transaction_fpds")
+                    transaction_fpds_sql = (
+                        "UPDATE transaction_fpds SET {} "
+                        "where detached_award_procurement_id = {}".format(
+                            pairs, load_object["transaction_fpds"]["detached_award_procurement_id"]
+                        )
+                    )
+                    cursor.execute(transaction_fpds_sql)
+
+                    columns, values, pairs = setup_load_lists(load_object, "transaction_normalized")
+                    load_object["transaction_fpds"]["transaction_id"] = results[0][0]
+                    load_object["award"]["latest_tranaction_id"] = results[0][0]
+                    transaction_normalized_sql = "UPDATE transaction_normalized SET {} " "where id  = '{}'".format(
+                        pairs, load_object["transaction_fpds"]["transaction_id"]
+                    )
+                    cursor.execute(transaction_normalized_sql)
+
+                    logger.debug("updated fpds transaction {}".format(results[0][0]))
+                else:
+                    # If there is no transaction we create a new one.
+                    # transaction_normalized and transaction_fpds should be one-to-one
+                    columns, values, pairs = setup_load_lists(load_object, "transaction_normalized")
+                    transaction_normalized_sql = "INSERT INTO transaction_normalized {} VALUES {} RETURNING id".format(
+                        columns, values
+                    )
+                    cursor.execute(transaction_normalized_sql)
+                    results = cursor.fetchall()
+                    load_object["transaction_fpds"]["transaction_id"] = results[0][0]
+                    load_object["award"]["latest_tranaction_id"] = results[0][0]
+
+                    columns, values, pairs = setup_load_lists(load_object, "transaction_fpds")
+                    transaction_fpds_sql = "INSERT INTO transaction_fpds {} VALUES {} RETURNING transaction_id".format(
+                        columns, values
+                    )
+                    cursor.execute(transaction_fpds_sql)
+                    results = cursor.fetchall()
+
+                    logger.debug("created fpds transaction {}".format(results[0][0]))
+
+                # No matter what, we need to go back and update the award's latest transaction to the award we just made
```

Comment:
Even if we could make this guarantee, I would still leave this logic out of here since we already go back and update awards on a second pass. Good catch

Reply:
Can you point me to the code that does that re-setting of the latest tx?

And if so, then we should run this update SQL to set latest transaction _only_ when a new award needs to be created, rather than as each and every transaction is processed.

Reply:
load_fpds_from_broker.py line 132 onward

Reply:
Well... that's not a very helpful reference. Please try to provide more precise pointers next time. 
Looks like:
- https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/etl/award_helpers.py#L49-L75
and
- https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/etl/award_helpers.py#L114

FYSA - looks like that code has changed in the last 4 days by Tony. Seems like cleanup rather than major changes.

----
Given that we always set/reset latest transaction, and also that I've found out `awards.latest_transaction_id` is a _nullable_ column, I think this whole section where we preemptively set the latest_transaction to the one being handled should probably be removed. Is there any further logic in this loader ETL that requires it to be set before the code referenced above does it?

Reply:
It doesn't matter for any of the code changed in this PR. I was trying to kill two birds with one stone by doing it in the same motion, but I'm pretty sure it's not worth it at this point.

Reply:
Resolving since the conversation seems complete and there are no additional requested code changes